cvmfs/sync_mediator.cc

```diff
@@ -94,6 +95,28 @@ void SyncMediator::Add(SharedPtr<SyncItem> entry) {
     return;
   }
 
+  // .cvmfsbundles file type
+  if (entry->IsBundleSpec()) {
+    std::string json_string;
```

Comment:
We should ensure that the file is a regular file (not a "special file") and not a hardlink.  Also that the file is in the root directory.  Otherwise we panic.